sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/TextOptions.scala

```diff
@@ -39,9 +39,12 @@ private[text] class TextOptions(@transient private val parameters: CaseInsensiti
    */
   val wholeText = parameters.getOrElse(WHOLETEXT, "false").toBoolean
 
+  val lineSeparator: String = parameters.getOrElse(LINE_SEPARATOR, "\n")
+  require(lineSeparator.nonEmpty, s"'$LINE_SEPARATOR' cannot be an empty string.")
 }
 
 private[text] object TextOptions {
   val COMPRESSION = "compression"
   val WHOLETEXT = "wholetext"
+  val LINE_SEPARATOR = "lineSep"
```

Comment:
+1

Reply:
It was taken after Univocity parser, `setLineSeparator`.

I think Hadoop calls it `textinputformat.record.delimiter` but the class name is `LineRecordReader` or `LineReader`.

Reply:
The name is not so matter as having the same name for the option across all supported datasources - text, csv and json to don't confuse users.

Reply:
To be clear, the name "lineSep" is taken after Python, R's `sep` and our supporting option `sep`. Here was another discussion - https://github.com/apache/spark/pull/18581#discussion_r134813986
"line" is taken after Univocity CSV parser and seems making sense in Hadoop too. I think we documented JSON is in lines, and I think it makes sense to CSV and text.

Reply:
Actually, I am not sure why we so care about this naming. I think all recordSeparator, recordSep, lineSeperator and lineSep are all fine actually. lineSep was choosen to be consistent at my best.

Reply:
In the example above, the line is counterintuitive for me. I imagine a line in text files as a sequence of one or more characters, displayed within a single horizontal sequence. I would prefer the short name *recSep* or *recordSeparator* for long name. I guess when the option will be used, it will separate text not by new line chars like `'\n'`, `'\r\n'`.

Reply:
One example might sound counterintuitive to you but it looks less consistent with other places at least I usually refer.

Reply:
I don't really care what we are going to call it this. I think it is important that we are consistent across datasources. IMO Since we can define anything to be a separator, not only newlines, `records` seems to fit a bit better.

Reply:
The definition of records was introduced in the classical paper: https://static.googleusercontent.com/media/research.google.com/en//archive/mapreduce-osdi04.pdf . I don't see any reasons for replacing it by **line** right now.

Reply:
My reason is to refer other places so that practically other users feel comfortable, which I usually put more importances. I really don't want to spend time on research why the other references used the term "line".

If we think about the plain text, CSV or JSON, the term "line" can be correct in a way. We documented http://jsonlines.org/ (even this reference used the term "line"). I think, for example, the line can be defined by its separator.

https://github.com/apache/spark/blob/c36fecc3b416c38002779c3cf40b6a665ac4bf13/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala#L1645

Reply:
We already used the term "line" everywhere in the doc. We could just say lines are separated by a character and minimise the doc fix and etc.